### DISCUSSION TITLE
SKU in Recently Viewed products

### DISCUSSION THREAD

**Discussion by pk-pankaj** [breezefront/community#114] [1. Help] [answered]
Hi @vovayatsyuk 

We created a module to achieve SKU in a recently viewed product and it is working fine with Luma but getting error when using Breeze

**Used js:**

 ```
define([
    'Magento_Ui/js/grid/columns/column',
    'Magento_Catalog/js/product/list/column-status-validator'
], function (Column, columnStatusValidator, $, stars) {
    'use strict';

    return Column.extend({
        /**
         * @returns {Boolean}
         */

        component: 'Magestudy_RecentlyViewed/js/product/new_one',

        isAllowed: function () {
            return columnStatusValidator.isValid(this.source(), 'new_one', 'show_attributes');
        },

        /**
         * @param row
         * @returns {string}
         */
        getValue: function (row) {
            return row['extension_attributes']['new_one'];
        }
    });
});
```

**Error in above js:**
`Uncaught TypeError: Cannot read properties of undefined (reading 'extend')`

Please check and help on this

Thanks,
Pankaj

**Comment by pk-pankaj** [comment]
@vovayatsyuk please look into this.

**Reply by vovayatsyuk** [reply]
We don't have a Column component. I will add documentation about recent products and let you know.

**Comment by pk-pankaj** [comment]
Thanks, @vovayatsyuk 

I saw the Column component in your module "Swissup_ProLabels" which you used with recently viewed products, so what alternative do you use to achieve that functionality with Breeze?

Thanks

**Reply by vovayatsyuk** [reply]
Here is how it's done in Prolabels: 

```js
(function () {
    'use strict';

    $.mixin('recentProducts', {
        getAdditionalContent: function (o, item, element) {
            if (item.extension_attributes && item.extension_attributes.swissup_prolabel) {
                $(element).prolabels(JSON.parse(item.extension_attributes.swissup_prolabel));
            }

            return o(item, element);
        }
    });
})();
```

**Reply by pk-pankaj** [reply]
Thanks, @vovayatsyuk 

Can you please give some assist how we can add SKUs in recently viewed products with Breeze?

**Reply by vovayatsyuk** [reply, marked as answer]
Have you tried something similar to this code in your js file:

```js
(function () {
    'use strict';

    $.mixin('recentProducts', {
        getAdditionalContent: function (o, item, element) {
            return (item.extension_attributes?.new_one || '') + o(item, element);
        }
    });
})();
```

Edit: Just added an article to the docs: https://breezefront.com/docs/recent-products

**Reply by pk-pankaj** [reply]
It worked, thanks @vovayatsyuk
